guacamole-common-js/src/main/webapp/modules/Tunnel.js

```diff
@@ -215,6 +215,24 @@ Guacamole.HTTPTunnel = function(tunnelURL, crossDomain) {
     var receive_timeout = null;
 
     /**
+     * Additional headers to be sent in tunnel requests. This dictionary can be
+     * populated with key/value header pairs to pass information such as authentication
+     * tokens, etc.
+     */
+    var extraHeaders = {};
```

Comment:
1. Declaring this via `var` produces a private variable, accessible only within the scope of this constructor. The `add_extra_headers()` function you've written will still work, but it's not accessing the `extraHeaders` declared here. If using this approach, `extraHeaders` will need to be defined as:

       this.extraHeaders = {};

2. Why not an optional parameter to the constructor, similar to `crossDomain`, rather than a public property?

Reply:
I just saw you mentioned:

> I've also preferred not to add the extra headers to the tunnel constructor to keep backward compatibility.

JavaScript will assign the value `undefined` to any parameter omitted from function calls, including constructors. As long as you ensure the behavior of the HTTP tunnel is identical to past releases in the case the parameter is omitted, this can be done without breaking compatibility.